tools/cli/commands/build.js

```diff
@@ -831,10 +859,37 @@ async function buildProject( t ) {
 		await fs.writeFile( `${ buildDir }/.gitattributes`, rules, { encoding: 'utf8' } );
 	}
 
+	// Get the project version number from the changelog.md file.
+	const projectVersionNumber = '';
+	try {
+		const rl = rlcreateInterface( {
+			input: createReadStream( `${ t.cwd }/CHANGELOG.md`, {
+				encoding: 'utf8',
+			} ),
+			crlfDelay: Infinity,
+		} );
+
+		// eslint-disable-next-line no-unused-vars
+		rl.on( 'line', line => {
+			// Todo: pull the correct version number from the changelog file.
+		} );
+		await once( rl, 'close' );
+
+		if ( ! projectVersionNumber ) {
+			throw new Error( 'Version number not found.' );
+		}
+	} catch ( err ) {
+		await t.setStatus( 'build failure' );
+		await t.output(
+			`\nError fetching latest version number from changelog.md while building project: ${ t.project }\n`
```

Comment:
Should make the message be accurate as to the filename, and no need to mention the project as it's already going to be added.
```suggestion
			`\nError fetching latest version number from CHANGELOG.md\n`
```
Or maybe even better, read the filename from composer.json (like `composerJson.extra?.changelogger?.[ 'changelog' ] || 'CHANGELOG.md'`) for use on both line 866 and here.

And maybe even better than that, use this text for the Error object (see the comment above) instead of printing it yourself.

Reply:
37e3508